src/cli/index.ts

```diff
@@ -0,0 +1,248 @@
+      }
+      process.exit(1);
+    }
+    return value;
+  }
+
+  process.stdout.write(chalk.underline.gray`Converting project:\n`);
+
+  const fullInputPath = resolve(process.cwd(), input);
+
+  const project = await writeStep(
+    `${chalk.bold("Importing")} ${inputType} project from path ${chalk.white(fullInputPath)}.`,
+    async () => {
+      let file: Buffer;
+      try {
+        file = fs.readFileSync(fullInputPath);
+      } catch (err) {
+        if (err instanceof Object && "code" in err) {
+          switch (err.code) {
+            case "ENOENT":
+              throw new StepError("File not found.");
+            case "EISDIR":
+              throw new StepError("Input path is a directory but should be a file.");
+          }
+        }
+        throw err;
+      }
+      const project = await Project.fromSb3(file);
+      return project;
+    }
+  );
+
+  switch (outputType) {
+    case "leopard": {
+      const leopard = await writeStep(`${chalk.bold("Converting")} project to ${chalk.white("Leopard")}.`, () => {
+        const leopard = project.toLeopard();
+        return leopard;
+      });
+
+      const fullOutputPath = resolve(process.cwd(), output);
+
+      await writeStep(`${chalk.bold("Exporting")} project to directory ${chalk.white(fullOutputPath)}.`, () => {
+        // First, make sure the output path is an empty directory (create it if it doesn't exist)
+        let existingFiles: string[];
+        try {
+          existingFiles = fs.readdirSync(fullOutputPath);
+        } catch (err) {
+          if (err instanceof Object && "code" in err) {
+            switch (err.code) {
+              case "ENOENT":
+                // Directory does not exist, create it
+                fs.mkdirSync(output, { recursive: true });
+                existingFiles = [];
+                break;
+              case "ENOTDIR":
+                throw new StepError("Output path is a file, not a directory.");
+              default:
+                throw err;
+            }
+          } else {
+            throw err;
+          }
+        }
+
+        if (existingFiles.length > 0) {
+          throw new StepError("Output directory is not empty.");
+        }
+
+        // Write the files
+        for (const [filename, contents] of Object.entries(leopard)) {
+          // Create directories as needed
+          fs.mkdirSync(resolve(fullOutputPath, filename, ".."), { recursive: true });
+
+          // Write the file
+          fs.writeFileSync(resolve(fullOutputPath, filename), contents);
+        }
+
+        for (const target of [project.stage, ...project.sprites]) {
+          for (const costume of target.costumes) {
+            const filename = `${target.name}/costumes/${costume.name}.${costume.ext}`;
+            const asset = Buffer.from(costume.asset as ArrayBuffer);
+            fs.mkdirSync(resolve(fullOutputPath, filename, ".."), { recursive: true });
+            fs.writeFileSync(resolve(fullOutputPath, filename), asset);
```

Comment:
You should be using `path.join` instead of hard-coding forward slashes (this might or might not work on Windows). And since you're actually building up the `filename` yourself here, you could store the directory separately and skip the `resolve(..., "..")` / `dirname(...)` call:

```suggestion
            const filedir = path.join(target.name, 'costumes');
            const filename = path.join(filedir, `${costume.name}.${costume.ext}`);
            const asset = Buffer.from(costume.asset as ArrayBuffer);
            fs.mkdirSync(resolve(fullOutputPath, filedir), { recursive: true });
```